src/merkle/merkle-tree.ts

```diff
@@ -62,7 +63,7 @@ export class MerkleTree<TData, TLeaf extends TData, TMetadata> {
     /**
      * Tree metadata
      */
-    readonly metadata: TMetadata
+    readonly metadata: TMetadata | null
```

Comment:
Can you add a comment about what it means that metadata is null?

Reply:
Added: "Tree metadata. Would not be considered for a proof if not available."